src/CarbonAware.WebApi/test/integrationTests/SciScoreControllerTests.cs

```diff
@@ -0,0 +1,112 @@
+namespace CarbonAware.WepApi.IntegrationTests;
+
+using System.Net;
+using NUnit.Framework;
+using System.Text.Json;
+using System.Net.Http.Headers;
+using WireMock.Server;
+using CarbonAware.Tools.WattTimeClient;
+
+/// <summary>
+/// Tests that the Web API controller handles and packages various responses from a plugin properly 
+/// including empty responses and exceptions.
+/// </summary>
+[TestFixture]
+public class SciScoreControllerTests
+{
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+	private APIWebApplicationFactory _factory;
+	private HttpClient _client;
+    private WireMockServer _server;
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    private bool resetEnvironment = false;
+
+    [OneTimeSetUp]
+    public void Setup()
+    {
+        //Defaults to JSON Integration Testing
+        //WattTime can be tested by uncommenting the line below or setting an env variable
+        //Environment.SetEnvironmentVariable("CarbonAwareVars__CarbonIntensityDataSource", "WattTime");
+
+        if (Environment.GetEnvironmentVariable("CarbonAwareVars__CarbonIntensityDataSource") == "WattTime")
```

Comment:
See the comments about the CarbonAwareController and requiring the environment variable to be set.